packages/plugins/GeoLocation/src/store/actions.ts

```diff
@@ -21,9 +23,35 @@ const actions: PolarActionTree<GeoLocationState, GeoLocationGetters> = {
         commit('setIsGeolocationDenied', true)
       }
     })
-    if (checkLocationInitially) {
+    if (getters.checkLocationInitially) {
       dispatch('track')
     }
+    dispatch('setupTooltip')
+  },
+  setupTooltip({ getters, rootGetters }) {
+    if (getters.showTooltip) {
+      const { map } = rootGetters
+      const overlay = new Overlay({
+        element: getTooltip(),
+        positioning: 'bottom-center',
+        offset: [0, -5],
+      })
+      map.addOverlay(overlay)
+      map.on('pointermove', ({ pixel, dragging }) => {
+        if (dragging) {
+          return
+        }
+        const features = map.getFeaturesAtPixel(pixel, {
+          layerFilter: (layer) =>
+            layer.get('name') === 'geoLocationMarkerLayer',
+        })
+
+        const coordinate = features.length
+          ? map.getCoordinateFromPixel(pixel)
+          : undefined
+        overlay.setPosition(coordinate)
+      })
+    }
   },
 
```

Comment:
```suggestion
    if (getters.showTooltip) {
      dispatch('setupTooltip')
    }
  },
  setupTooltip({ rootGetters }) {
    const { map } = rootGetters
    const overlay = new Overlay({
      element: getTooltip(),
      positioning: 'bottom-center',
      offset: [0, -5],
    })
    map.addOverlay(overlay)
    map.on('pointermove', ({ pixel, dragging }) => {
      if (dragging) {
        return
      }
      const features = map.getFeaturesAtPixel(pixel, {
        layerFilter: (layer) => layer.get('name') === 'geoLocationMarkerLayer',
      })

      const coordinate = features.length
        ? map.getCoordinateFromPixel(pixel)
        : undefined
      overlay.setPosition(coordinate)
    })
  },
```
The action doesn't have to be called if the `showTooltip` is set to `false`

Reply:
I'd like the action to decide for itself.